contentcuration/contentcuration/serializers.py

```diff
@@ -68,17 +69,23 @@ def get(*args, **kwargs):
     def get_file_url(self, obj):
         return obj.file_on_disk.url
 
+    def get_storage_path(self, obj):
+        return generate_storage_path(obj.checksum, obj.checksum + '.' + obj.file_format.extension)
+
     def retrieve_recommended_kind(self, obj):
-        if obj.contentnode and obj.contentnode.kind:
+        if obj.contentnode is not None and obj.contentnode.kind:
             return obj.contentnode.kind.pk
-        return FormatPreset.objects.filter(allowed_formats__extension__contains=obj.file_format).first().kind.pk
+        preset = FormatPreset.objects.filter(allowed_formats__extension__contains=obj.file_format.extension).first()
```

Comment:
Similar to the email queries: Doesn't need `__contains`, just something like `allowed_formats__extension=`